SSSD/sssd
# passkey: register key with helper process

Create a helper process to handle the passkey key registration. In order
to do that the helper process will parse the command line arguments,
list the connected devices, select the first one in the list, prepare
the user credentials, generate the key credentials, verify them to
ensure that everything has worked correctly and print the credentials.

FIDO U2F fallback has been disabled to force a PIN usage, thus
increasing the security as a lost or stolen key would be unusable with
it.

Moreover, include the libfido2 dependency in the configuration and the
spec file.

Finally, several test cases have been added to the unit tests. Most of
them just check the happy path but some error handling is also tested.

The output contains the passkey keyword, the credential ID and the
public key. Example:
`passkey:9S87qLk8/RxYJ3skwwYduomAM+/HDtz41N0+w/vRL6aGKJkLMsg+2OhO0E8pK5DuO1KmdK61K8PmH7jiYuOqbg==,9YE1s/f7J47h2A/DXCVFWulqoBXFzCSxcbGEBadkpSUFjwUudhPLnPUTv2qNamakXJgRYCZQ7vpS/t5zXMLnkw==`

Resolves: https://github.com/SSSD/sssd/issues/6228

## How to test

The words in bold should be replaced by a real value.

Generate the key registration
$ ./passkey_child --register --username=**user** --domain=**test.com**

The following options can also be set when executing the previous command to print debugging logs
--debug-level=9 --logger=stderr

Set client data hash
$ echo assertion challenge | openssl sha256 -binary | base64 > test_param

Set relying party id (use the same value as in --domain)
$ echo **test.com** >> test_param

Set credential id (printed in as the output of the first step, between `passkey:` and `,`)
$ echo **credentialId** >> test_param

Obtain assertion (check dmesg to see what hidraw interface has been set)
$ fido2-assert -G -i test_param **/dev/hidrawX**
